stl/inc/algorithm

```diff
@@ -1789,6 +1789,52 @@ namespace ranges {
     };
 
     inline constexpr _Move_fn move{_Not_quite_object::_Construct_tag{}};
+
+    // ALIAS TEMPLATE move_backward_result
+    template <class _In, class _Out>
+    using move_backward_result = in_out_result<_In, _Out>;
+
+    // VARIABLE ranges::move_backward
+    class _Move_backward_fn : private _Not_quite_object {
+    public:
+        using _Not_quite_object::_Not_quite_object;
+
+        // clang-format off
+        template <bidirectional_iterator _It1, sentinel_for<_It1> _Se1, bidirectional_iterator _It2>
+            requires indirectly_movable<_It1, _It2>
+        constexpr move_backward_result<_It1, _It2> operator()(_It1 _First, _Se1 _Last, _It2 _Result) const {
+            _Adl_verify_range(_First, _Last);
+            auto _UFirst = _Get_unwrapped(_STD move(_First));
+            auto _ULast  = _Get_final_iterator_unwrapped<_It1>(_UFirst, _STD move(_Last));
+            _Seek_wrapped(_First, _ULast);
+            _Result = _Move_backward_common(_STD move(_UFirst), _STD move(_ULast), _STD move(_Result));
+            return {_STD move(_First), _STD move(_Result)};
+        }
+
+        template <bidirectional_range _Rng, bidirectional_iterator _It>
+            requires indirectly_movable<iterator_t<_Rng>, _It>
+        constexpr move_backward_result<borrowed_iterator_t<_Rng>, _It> operator()(_Rng&& _Range, _It _Result) const {
+            auto _ULast = _Get_final_iterator_unwrapped(_Range);
+            _Result     = _Move_backward_common(_Ubegin(_Range), _ULast, _STD move(_Result));
+            return {_Rewrap_iterator(_Range, _STD move(_ULast)), _STD move(_Result)};
+        }
+        // clang-format on
+    private:
+        template <class _It1, class _It2>
+        _NODISCARD static constexpr _It2 _Move_backward_common(const _It1 _First, _It1 _Last, _It2 _Result) {
+            _STL_INTERNAL_STATIC_ASSERT(bidirectional_iterator<_It1>);
+            _STL_INTERNAL_STATIC_ASSERT(bidirectional_iterator<_It2>);
+            _STL_INTERNAL_STATIC_ASSERT(indirectly_movable<_It1, _It2>);
+
+            while (_First != _Last) {

```

Comment:
In `ranges::copy_backward` you did include the `std::optimizations` by utilizing `_Copy_backward_unchecked`.

I assume you cannot do that here because that algorithm uses `std::iter_move` which does not work well with proxies. Would it make sense to add optimization anyway?

Reply:
> Would it make sense to add optimization anyway?

Sure - done.